rpc/types_broadcast_transaction.go

```diff
@@ -21,15 +21,20 @@ var (
 	_ BroadcastInvokeTxnType = BroadcastInvokev3Txn{}
 )
 
-type BroadcastDeclareTxnType interface{}
+type BroadcastDeclareTxnType interface {
+	GetContractClass() interface{}
+}
 
 var (
 	_ BroadcastDeclareTxnType = BroadcastDeclareTxnV1{}
 	_ BroadcastDeclareTxnType = BroadcastDeclareTxnV2{}
 	_ BroadcastDeclareTxnType = BroadcastDeclareTxnV3{}
 )
 
-type BroadcastAddDeployTxnType interface{}
+type BroadcastAddDeployTxnType interface {
+	GetConstructorCalldata() []*felt.Felt
+	GetContractAddressSalt() *felt.Felt
```

Comment:
nit: since the purpose of these methods are to contrain the interface, I think we only need one of them, so imo we can remove (eg) GetContractAddressSalt

Reply:
Sure,
Done that and rebased the PR...
ready to be merged